Iteration 3 movie details

pshim17 opened the pull request:
Type of change

  - [X] New feature
  - [ ] Bug Fix

Check the correct boxes

  - [X] This code broke nothing
  - [ ] This code broke some stuff
  - [ ] This code broke everything

Implements/Fixes:

- Added onClick to be able to click on each poster and display movie details. 
- Added appropriate movie details to display: backdrop image, genres, and description 
- Added 'home icon' to the header; returns home when clicked on. 
- Only one movie detail is displayed, header is still there showing title and home icon. 
- Home icon is removed when showing movies container/poster not clicked on Added styling to match the previews 
- Added pointer cursor when hovering over poster.
- 
Testing Changes:

  - [X] I have fully tested my code 
  - [ ] All tests are passing
  - [ ] Some tests are failing
  - [ ] All tests are failing

  - [ ] No previous tests have been changed
  - [ ] Some previous tests have been changed
  - [ ] All of the previous tests have been changed (Please describe what in the world happened that all of the previous tests needed changing.)

Checklist:

  - [ ] I have reviewed my code
  - [ ] The code will run locally
  - [ ] My code has no unused/commented out code
  - [ ] I have commented my code, particularly in hard-to-understand areas

(Optional) What questions do you have? Anything specific you want feedback on?

  * I ran into an issue where when I load the page and hover over each poster, it took me a long time to get the pointer cursor to show. Not sure if this is just an issue I'm experiencing and was wondering if it also happening to you?
  * I just picked colors that look similar to what we see on the preview, we can come back and change the colors later if you want to make it look better!

(For Fun!) Please include a link to a gif [or add an emoji] of your feelings about this branch.

Link:

Turing School of Software and Design, a Colorad

pshim17 committed:
feat: added home icon in MovieDetails to return back to home when clicking on the icon

pshim17 committed:
fix: fixed github uploading issue

JonoSommers commented on src/MovieDetails/MovieDetails.js:
```diff
@@ -1,9 +1,16 @@
   return (
     <section className='MovieDetails'>
-      <p>Movie Details go here!</p>
+      <img src= { movie.backdrop_path } alt={ movie.title }/>
```
should we include 'poster' at the end of the alt text? It will make it clear that if the picture does not render, that it should be the movie poster.

pshim17 replied:
Sounds good with me! We should make that change. I agree, adding "poster" at the end of the alt text will improve clarity and accessibility!

JonoSommers commented on src/MoviePoster/MoviePoster.js:
```diff
@@ -1,10 +1,10 @@
   return (
     <section className='MoviePoster'>
-      <img src= { poster_path } alt= "Movie Poster"/>
+      <img src= { poster_path } alt= "Movie Poster" onClick={onClick}/>
```
I could be wrong, but should this be onClick={handleClick}? If not, could you explain what this snippet is doing?

pshim17 replied:
That is a great point you made. I was passing the handleClick function down to MoviePoster as onClick in the parameter. I can definitely see why it could be confusing. We should def make this change so that I'm passing in handleClick instead of onClick! Thanks for point that out!

JonoSommers reviewed (commented):
Over all everything looks great! Before I merge, if you could drop a response to my comments when you get a moment that would be awesome, thank you!

pshim17 committed:
Merge branch 'main' into iteration-3-movie-details

JonoSommers committed:
Update MovieDetails.js

Add poster to the alt text for img src

JonoSommers committed:
Update MoviePoster.js

changed on onClick prop value to handleClick to follow better naming conventions

JonoSommers committed:
Update MoviesContainer.js

changing to handleClick again

JonoSommers committed:
Merge branch 'main' into iteration-3-movie-details
